### ISSUE TITLE
Warning on task deploy: "aws-cdk-lib.aws_lambda.FunctionOptions#logRetention is deprecated."

### ISSUE THREAD

**Issue by vboufleur** [globaldatanet/aws-firewall-factory#293]
**Describe the bug**
While deploying the stack the following warning appears:
```
[WARNING] aws-cdk-lib.aws_lambda.FunctionOptions#logRetention is deprecated.
  instead create a fully customizable log group with `logs.LogGroup` and use the `logGroup` property to instruct the Lambda function to send logs to it.
Migrating from `logRetention` to `logGroup` will cause the name of the log group to change.
Users and code and referencing the name verbatim will have to adjust.

In AWS CDK code, you can access the log group name directly from the LogGroup construct:
"
declare const myLogGroup: logs.LogGroup;
myLogGroup.logGroupName;
"
  This API will be removed in the next major release.
```

**How to reproduce it**
Just deploy a stack.

**Expected behaviour**
No warnings, as they indicate that in the next major CDK release the deployment will break.

**Please fill in the following information about the solution:**.
- Version: 4.2.1
- Region: us-east-1
- Has the solution been modified from the version posted to this repository? Yes
- If yes, are the changes available on GitHub? No

**Screenshots**.
![image](https://github.com/globaldatanet/aws-firewall-factory/assets/16310611/06a00c82-a53f-4666-92e8-9f099ae54484)

**Comment by daknhh** [comment]
Thanks for adding the bug, i will fix this in the next version

**Comment by daknhh** [comment]
fixed in 4.2.3
